marimo/_plugins/ui/_impl/file_browser.py

```diff
@@ -218,33 +218,47 @@ def _list_directory(
                 "Navigation is restricted; navigating to a "
                 "parent of initial path is not allowed."
             )
-
+        folders: list[TypedFileBrowserFileInfo] = []
         files: list[TypedFileBrowserFileInfo] = []
+
         for file in path.iterdir():
```

Comment:
What is the list order of iterdir? Is it automatically alphabetized?

Think there's a weird edgecase where `> self._limit` starts to cause weird things.

e.g.

c
d
--- # limit
a
b

`a` `b` are never seen, despite maybe being first

Maybe this is fine. I think GH prompts that list orderings may be invalid for > 1000 files- but also maybe not a super common occurrence